crates/pathfinder/src/sequencer.rs

```diff
@@ -1,37 +1,27 @@
 //! StarkNet L2 sequencer client.
+mod builder;
 pub mod error;
 pub mod reply;
 pub mod request;
 
-use self::{
-    error::StarknetError,
-    request::{add_transaction::ContractDefinition, Call},
-};
+use self::request::{add_transaction::ContractDefinition, Call};
 use crate::{
     core::{
-        CallSignatureElem, ClassHash, ConstructorParam, ContractAddress, ContractAddressSalt, Fee,
-        StarknetTransactionHash, StorageAddress, StorageValue, TransactionNonce,
-        TransactionVersion,
+        BlockId, CallSignatureElem, ClassHash, ConstructorParam, ContractAddress,
+        ContractAddressSalt, Fee, StarknetTransactionHash, StorageAddress, StorageValue,
+        TransactionNonce, TransactionVersion,
     },
     ethereum::Chain,
-    rpc::types::{BlockHashOrTag, BlockNumberOrTag, Tag},
+    rpc::types::BlockHashOrTag,
     sequencer::error::SequencerError,
 };
 use reqwest::Url;
-use std::{borrow::Cow, fmt::Debug, future::Future, result::Result, time::Duration};
+use std::{fmt::Debug, result::Result, time::Duration};
 
 #[cfg_attr(test, mockall::automock)]
 #[async_trait::async_trait]
 pub trait ClientApi {
-    async fn block_by_number(
-        &self,
-        block_number: BlockNumberOrTag,
-    ) -> Result<reply::Block, SequencerError>;
-
-    async fn block_by_hash(
-        &self,
-        block_hash: BlockHashOrTag,
-    ) -> Result<reply::Block, SequencerError>;
+    async fn block(&self, block: BlockId) -> Result<reply::Block, SequencerError>;
```

Comment:
I wanted the following:
```suggestion
    async fn block<B: Into<BlockId>>(&self, block: B)-> Result<reply::Block, SequencerError>;
```
but unfortunately this does not work with the mock library. Mostly just means that callers must append `.into()` themselves on some calls. Might revisit this if / when we remove mocks for channels.

Reply:
It is better probably overall but this probably would had removed the object safety from the method, which is why it cannot be used. In general there has recent-ishly been discussion on the amount of bloat this pattern creates and what kind of countermeasures it needs, for example some stdlib functions are constructed with an generic layer on top of unsized reference accepting arg: https://doc.rust-lang.org/stable/src/std/fs.rs.html#228-236